File: JvmClient/src/commonMain/kotlin/spaceEngineers/navigation/NavGraph.kt
package spaceEngineers.navigation

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import spaceEngineers.graph.DirectedGraph
import spaceEngineers.model.BlockId
import spaceEngineers.model.Vec3F

@Serializable
data class NavGraph(
    @SerialName("Nodes")
    val nodes: List<Node> = emptyList(),
    @SerialName("Edges")
    val edges: List<Edge> = emptyList()
) {
    fun toDirectedGraph(): DirectedGraph<BlockId, Vec3F, String, String> {
        return DirectedGraph(
            nodes = this.nodes,
            edges = this.edges,
        )
    }
}

File: Source/Ivxr.SpaceEngineers/Navigation/NavGraph.cs
using System.Collections.Generic;

namespace Iv4xr.SpaceEngineers.Navigation
{
    public class NavGraph
    {
        public readonly List<Node> Nodes;

        public readonly List<Edge> Edges;

        public NavGraph(List<Node> nodes, List<Edge> edges)
        {
            Nodes = nodes;
            Edges = edges;
        }
    }
}
